packages/sdk/src/resolve/resolve-params.js

```diff
@@ -1,14 +1,23 @@
-import {isTransaction, isScript, pipe, get, Ok, update} from "@onflow/interaction"
+import {isTransaction, isScript, get, Ok} from "@onflow/interaction"
 
-function isFn(v) {
-  return typeof v === "function"
-}
+const isFn = d => typeof d === "function"
+const isString = s => typeof s === "string"
 
-export const resolveParams = pipe([
-  ix => {
-    if (!isTransaction(ix) && !isScript(ix)) return Ok(ix)
-    const code = get(ix, "ix.code")
-    ix.payload.code = isFn(code) ? code(get(ix, "ix.params", {})) : code
+export const resolveParams = (ix) => {
+  if (!(isTransaction(ix) || isScript(ix))) return Ok(ix)
+  const cadence = get(ix, 'ix.cadence')
+  if (isString(cadence)) {
+    ix.message.cadence = cadence
     return Ok(ix)
   }
-])
+  if (isFn(cadence)) {
+    const params = Object.fromEntries(Object
+      .values(ix.params)
```

Comment:
these values can be an async function, can you make sure they get resolved before the filter and transform